### PR TITLE
Added `iter` methods to `ObjectList`

### PR DESCRIPTION
**Pull request by kitfre**
Added `iter` and `iter_mut` methods to `ObjectList` to provide Iterators, as well as implementations of `IntoIterator` for `ObjectList`, `&ObjectList` and `&mut ObjectList`

#41

### CODE REVIEW

**Comment by clux** [comment]
This is solid. Examples are really good, code is clean. Just a few typos in there that causes CI to fail atm.

Will definitely merge this when it passes, but I just have a small request:

change the pod_openapi.rs example to use the simpler iterator. it's currently doing a `for p in pod.list().items`.!

File: src/api/mod.rs [hunk 1]
```diff
@@ -34,24 +26,14 @@ pub use typed::{
...
 #[cfg(feature = "openapi")]
 mod openapi;
 #[cfg(feature = "openapi")]
 mod snowflake;
 #[cfg(feature = "openapi")]
-pub use snowflake::{v1Event, v1Secret, v1ConfigMap};
+pub use snowflake::{v1ConfigMap, v1Event, v1Secret};
 
 mod metadata;
-pub use self::metadata::{
```
Comment (kitfre) [inline src/api/mod.rs]: I had to add `ListMeta` to the public exports for use in the example in the docs on the new `iter` methods, is this okay or should I just cut out the examples?

Comment (clux) [reply src/api/mod.rs]: that should probably be public anyway :+1:

thanks for pointing it out though. the re-formatting made that hard to spot.

**Comment by clux** [comment]
Released in kube 0.20.0 :-)

**Comment by kitfre** [comment]
Awesome thanks! There's a good chance I pick at a couple other issues as well, rust and kuberentes are two of my favorite things and this project is awesome 🦀

**Pull request merged**